test/api/v3/integration/groups/POST-groups_groupId_leave.test.js

```diff
@@ -338,4 +340,48 @@ describe('POST /groups/:groupId/leave', () => {
       });
     });
   });
+
+  each(typesOfGroups, (groupDetails, groupType) => {
+    context(`Leaving a group with extraMonths left plan when the group is a ${groupType}`, () => {
+      const extraMonths = 12;
+      let groupWithPlan;
+      let member;
+
+      beforeEach(async () => {
+        const { group, members } = await createAndPopulateGroup({
+          groupDetails,
+          members: 1,
+          upgradeToGroupPlan: true,
+        });
+        [member] = members;
+        groupWithPlan = group;
+        await member.update({
+          'purchased.plan.extraMonths': extraMonths,
+        });
+      });
+      it('calculates dateTerminated and sets extraMonths to zero after user leaves the group', async () => {
+        const userBeforeLeave = await User.findById(member._id).exec();
+
+        await member.post(`/groups/${groupWithPlan._id}/leave`);
+        const userAfterLeave = await User.findById(member._id).exec();
+
+        const dateTerminatedBefore = userBeforeLeave.purchased.plan.dateTerminated;
+        const extraMonthsBefore = userBeforeLeave.purchased.plan.extraMonths;
+        const dateTerminatedAfter = userAfterLeave.purchased.plan.dateTerminated;
+        const extraMonthsAfter = userAfterLeave.purchased.plan.extraMonths;
+
+        const expectedTerminationDate = calculateSubscriptionTerminationDate(null, {
```

Comment:
about this check, we know that the server will call `calculateSubscriptionTerminationDate` so if we check the server result against the `calculateSubscriptionTerminationDate` call we risk not catching  issues with `calculateSubscriptionTerminationDate`. Could we change `expectedTerminationDate` to a static value that can be checked against the server result?

If it becomes difficult to know the exact date we can relax the check at line 383 to return true even if there's a slight difference

Reply:
@paglias I think any issues with the `calculateSubscriptionTerminationDate` function should be resolved by the unit tests I wrote (this function is pure)

Reply:
Ok but if the functions returned a termination date in 2050 we wouldn't notice because `dateTerminatedAfter` and `expectedTerminationDate` are both calculated the same way. We should either have unit tests on `calculateSubscriptionTerminationDate` or the integration test should be `we know that this subscription termination date should be set in ~12 months from now`, after calling the `leave` route it's set to the correct value? Does it make sense?

Reply:
> We should either have unit tests on `calculateSubscriptionTerminationDate` or the integration test should be...

Hmm, but we **do** have unit tests on `calculateSubscriptionTerminationDate`, it's in the same commit, first file

Reply:
sorry, you're right I don't know how I missed